packages/dds/tree/src/core/edit-manager/editManager.ts

```diff
@@ -124,7 +124,7 @@ export class EditManager<
     }
 
     public getTrunk(): readonly RecursiveReadonly<Commit<TChangeset>>[] {
-        return this.trunk;
+        return this.trunk as RecursiveReadonly<Commit<TChangeset>>[];
```

Comment:
I didn't expect to have to make a type assertion here. Without it the compiler complains that Type 'TChangeset' is not assignable to type 'RecursiveReadonly<TChangeset>'.
Same below for `getLocalChanges`.
Any idea what's going on?

Reply:
typescript is unable to prove that for all possible type T, T is assignable to RecursiveReadonly<T>. If you add `T & ` to the beginning of the definition to RecursiveReadonly, it fixes that.

This does not fix getLocalChanges below unless you move the `[]` inside the `RecursiveReadonly`, and if you don't it gives gives errors that don't make sense to me.